native/src/main/java/io/ballerina/stdlib/data/xml/XmlParser.java

```diff
@@ -0,0 +1,853 @@
+            ((BMap<BString, Object>) currentElement).put(StringUtils.fromString(Constants.CONTENT), result);
+        } else {
+            currentNode.put(currentFieldName, result);
+        }
+    }
+
+    private QualifiedName getLastElementInSiblings(LinkedHashMap<QualifiedName, Boolean> siblings) {
+        Object[] arr = siblings.keySet().toArray();
+        QualifiedName lastElement = null;
+        if (arr.length != 0) {
+            lastElement = (QualifiedName) arr[arr.length - 1];
+        }
+        return lastElement;
+    }
+
+    @SuppressWarnings("unchecked")
+    private Map<QualifiedName, Field> getAllFieldsInRecordType(RecordType recordType, XmlParserData xmlParserData) {
+        BMap<BString, Object> annotations = recordType.getAnnotations();
+        HashMap<String, QualifiedName> modifiedNames = new LinkedHashMap<>();
+        for (BString annotationKey : annotations.getKeys()) {
+            String keyStr = annotationKey.getValue();
+            if (keyStr.contains(Constants.FIELD)) {
+                // Capture namespace and name from the field annotation.
+                String fieldName = keyStr.split(Constants.FIELD_REGEX)[1];
+                Map<BString, Object> fieldAnnotation = (Map<BString, Object>) annotations.get(annotationKey);
+                QualifiedName fieldQName = DataUtils.getFieldNameFromRecord(fieldAnnotation, fieldName);
+                fieldQName.setLocalPart(getModifiedName(fieldAnnotation, fieldName));
+                modifiedNames.put(fieldName, fieldQName);
+            }
+        }
+
+        Map<QualifiedName, Field> fields = new HashMap<>();
+        Map<String, Field> recordFields = recordType.getFields();
+        for (String key : recordFields.keySet()) {
+            QualifiedName modifiedQName =
+                    modifiedNames.getOrDefault(key, new QualifiedName(QualifiedName.NS_ANNOT_NOT_DEFINED, key, ""));
+            if (fields.containsKey(modifiedQName)) {
+                throw DiagnosticLog.error(DiagnosticErrorCode.DUPLICATE_FIELD, modifiedQName.getLocalPart());
+            } else if (xmlParserData.attributeHierarchy.peek().containsKey(modifiedQName)) {
+                continue;
+            }
+
+            fields.put(modifiedQName, recordFields.get(key));
+        }
+        return fields;
+    }
+
+    @SuppressWarnings("unchecked")
+    private Map<QualifiedName, Field> getAllAttributesInRecordType(RecordType recordType) {
+        BMap<BString, Object> annotations = recordType.getAnnotations();
+        Map<QualifiedName, Field> attributes = new HashMap<>();
+        for (BString annotationKey : annotations.getKeys()) {
+            String keyStr = annotationKey.getValue();
+            if (keyStr.contains(Constants.FIELD) && DataUtils.isAttributeField(annotationKey, annotations)) {
+                String attributeName = keyStr.split("\\$field\\$\\.")[1].replaceAll("\\\\", "");
```

Comment:
Unless you have a check to ensure that the record will contain `$field$`, this may cause an `ArrayIndexOutOfBoundsException` exception.

Reply:
This is for field annotation and there is a check  `keyStr.contains(Constants.FIELD)` already there to ensure the prefix is presented in the string